src/protocols/ethereum/views/TransactionDetails.vue

```diff
@@ -123,10 +132,24 @@ export default defineComponent({
     const fee = computed((): number => transaction.value?.tx?.fee || 0);
     const amount = computed((): number => transaction.value?.tx?.amount || 0);
 
+    async function getDecodedCallData() {
+      if (innerTx.value?.callData && innerTx.value?.recipientId) {
+        return decodeTxData(
+          innerTx.value.callData,
+          innerTx.value.recipientId,
+          activeAccount.value.address,
+        );
+      }
+      return null;
+    }
+
     watch(
       transaction,
       (value) => {
         setLoaderVisible(!value);
+        getDecodedCallData().then((decodedData) => {
+          decodedCallData.value = decodedData;
+        });
```

Comment:
```suggestion
        decodedCallData.value  = await getDecodedCallData();
```

And add `async` before `(value) =>`